### DISCUSSION TITLE
Override dependencies that dont return nothing

### DISCUSSION THREAD

**Discussion by Soufraz** [fastapi/fastapi#6717] [Questions] [answered]
Hi.. I started this discussion on discord but seems like a more tricky thing.

I'd like to know if dependency_overrides works when the dependency function dont return anything.. in my case my dependency function only pass if not wrong happens or throw an httpexception if something wrong happens..
The dependency override is not working because in fastapi.dependencies.utils, when the code will try override the dependencies, the reference of functions are different and then the override dont occurs..

My dependency:
```
async def database_check():
    try:
        _database_is_alive()
        if not database.is_connected:
            await database.connect()

    except ConnectionDatabaseError:
        if database.is_connected:
            await database.disconnect()

        raise HTTPException(status_code=503)
```

My 3 attempts functions to override dependency:
```
from ..api.dependencies import database_check

async def database_check_pass():
    return True

async def database_check_pass():
    pass

async def database_check_pass():
    def handle():
      return True
    return handle
```

My override dependency code:
`app.dependency_overrides[database_check] = database_check_pass`

Something I notice, inside fastapi code, the function reference inside dependency_overrides_provider is different than the original function in app/api/dependencies.py

fastapi.dependencies.utils
```
call = getattr(
    dependency_overrides_provider, "dependency_overrides", {}
).get(original_call, original_call)
```

My app structure:
```
app/
  api/
    dependencies.py
    views.py
  tests/
    my_test.py
  main.py
```

What am I missing here?

**Comment by Mause** [comment]
I can't see any issues with this code - here's a full example that works fine:


```python
from fastapi import FastAPI, Depends
from fastapi.testclient import TestClient


app = FastAPI()


async def database_check():
    try:
        _database_is_alive()
        if not database.is_connected:
            await database.connect()

    except ConnectionDatabaseError:
        if database.is_connected:
            await database.disconnect()

        raise HTTPException(status_code=503)


async def database_check_pass():
    return "the value were expecting"


app.dependency_overrides[database_check] = database_check_pass


@app.get("/")
def route(database_check=Depends(database_check)):
    return database_check


tc = TestClient(app)
assert tc.get("/").json() == "the value were expecting"
```

**Comment by Soufraz** [comment]
@Mause thank you a lot for trying to help me.
But you didnt replicate my case. You made a fresh new one. Without packages and imports.
Try to reply the same you did but within this structure:

```
app/
  api/
    dependencies.py
    views.py
  tests/
    my_test.py
  main.py
```

:)

**Comment by Kludex** [comment]
Why don't you provide that structure within a repository, so we can help you easily?

**Comment by Soufraz** [comment]
@Kludex 
Right. I'll do that. 
When ready I'll leave the link here. 
Thank you.

**Comment by Soufraz** [comment]
Copying only dependencies files and routes to a fresh project works properly.  :|
I'll move file by file to this new fresh project until get the error.
Is there some cache to clean or anything related?

**Comment by Soufraz** [comment]
I closed because probably is something related to my project.

**Comment by Soufraz** [comment, marked as answer]
Found! In tests file I was importing ..main and ..api.dependencies
Changed to app.main and app.api.dependencies.
Thank you guys!
